### ISSUE TITLE
Support epoch based validation early stopping

### ISSUE THREAD

**Issue by ecly** [modernmt/modernmt#508]
Currently, the early stopping is implemented by looking at the TensorboardX loss at every N for `--save-interval-updates N`. This results in unexpected behavior if one tries to disable interval-based checkpoints, and want to stick with epoch-based checkpoints (similar to default fairseq behavior). Namely it will cause a division by zero error at: https://github.com/modernmt/modernmt/blob/master/cli/train.py#L81

Basically, this renders what I would consider a fairly intutive combination of options broken:
```bash
--save-interval-updates 0 \
--save-interval 1 \
--validate-interval 1
```

One possible easy fix (perhaps not pretty) might be to derive the value for `save_interval_updates` used in the early-stopping code from the other parameters.

**Comment by nicolabertoldi** [comment]
@ecly 

We will take into consideration your request.
Stay tuned!

**Comment by ecly** [comment]
This could be trivially added when in the future you update the version of `fairseq` as they have added native support for early stopping on the master branch (also means you could get rid of your current somewhat hacky implementation thereof). 
They have however yet to make a release after the merge, but I imagine that will come shortly.
